airbyte-integrations/connectors-performance/destination-harness/src/main/java/io/airbyte/integrations/destination_performance/PerformanceTest.java

```diff
@@ -0,0 +1,202 @@
+    this.destination = new DefaultAirbyteDestination(dstIntegtationLauncher);
+    final var jobRoot = "/";
+    log.info("Destination starting");
+    destination.start(dstConfig, Path.of(jobRoot));
+
+    // Try read logs.
+    final var logListener = CompletableFuture.runAsync(() -> {
+      log.info("Listening to destination logs");
+      while (!destination.isFinished()) {
+        final Optional<AirbyteMessage> messageOptional;
+        try {
+          messageOptional = destination.attemptRead();
+          messageOptional.ifPresent(airbyteMessage -> log.info("dst log: {}", airbyteMessage));
+
+        } catch (final Exception e) {
+          throw new DestinationException("Destination process read attempt failed", e);
+        }
+      }
+    });
+
+    log.info("Get datasource {}", this.dataSourceURL);
+    Path temp = Files.createTempFile("", ".tmp");
+    final URL url = new URL(this.dataSourceURL.trim());
+    ReadableByteChannel readableByteChannel = Channels.newChannel(url.openStream());
+    FileOutputStream fileOutputStream = new FileOutputStream(temp.toString());
+    FileChannel fileChannel = fileOutputStream.getChannel();
+    final var bytes = fileChannel.transferFrom(readableByteChannel, 0, Long.MAX_VALUE);
+    readableByteChannel.close();
+    fileOutputStream.close();
+    log.info("done saving datasource {} ({})", temp.toString(), bytes);
+    BufferedReader reader = new BufferedReader(new FileReader(temp.toString()));
+
+    log.info("Reading datasource header");
+    final var columnsString = reader.readLine();
+    log.info("header line: {}", columnsString);
+    final Pattern pattern = Pattern.compile(",");
+    final var columns = Arrays.asList(pattern.split(columnsString));
+    var totalBytes = 0.0;
+    var counter = 0L;
+    final var start = System.currentTimeMillis();
+
+    try (reader) {
+      while (true) {
+        String line = reader.readLine();
+        if (line == null) {
+          log.info("End of datasource after {} lines", counter);
+          break;
+        }
+        final List row;
+        try {
+          row = Arrays.asList(pattern.split(line));
+        } catch (NullPointerException npe) {
+          log.warn("Bad line: {} {} {}", line, counter, totalBytes);
+          continue;
+        }
+        assert (row.size() == columns.size());
+        StringBuilder sb = new StringBuilder();
+        sb.append("{");
+        Iterator<String> rowIterator = row.iterator();
+        Iterator<String> colIterator = columns.iterator();
+        ArrayList<String> combined = new ArrayList<>(columns.size());
+        while (colIterator.hasNext() && rowIterator.hasNext()) {
+          combined.add("\"%s\":\"%s\"".formatted(colIterator.next(), rowIterator.next()));
+        }
+        sb.append(String.join(",", combined));
+        sb.append("}");
+        final String recordString = sb.toString();
+        // log.info("*** RECORD: {}", recordString); // TEMP
+        totalBytes += recordString.length();
```

Comment:
When we add the record to AirbyteMessage with `.withData(Jsons.deserialize(recordString)))` 
We deserializes this json.
But it will be again serialized when written out to destination in `destination.accept(airbyteMessage)`, 
And counting on platform is done on the serialized number of bytes - that is the string representation of the record json - which is what I'm counting here.

Reply:
from https://github.com/airbytehq/airbyte/pull/25030/files#r1194376651, this looks correct to me, however it's easy to misunderstand (as per Ryan's comment), since it's alot of lines together.